<?xml version="1.0"?>
<launch>
    <arg name="nao_ip" default="127.0.0.1"/>
    <arg name="shm_id1" default="img1"/>
    <arg name="shm_id2" default="img2"/>
    <arg name="res" default="1" /> <!--1 for QVGA, 2 for VGA-->
    <arg name="topic1" default="cam0" />
    <arg name="topic2" default="cam1" />
    <arg name="frame" default="world" />

    <!-- Launch node on ros end to publish image from shared memory -in shared memory -->
    <node pkg="rosnao_bridge" type="image_relay" name="image_relay1" output="screen" args="$(arg shm_id1) $(arg res) $(arg topic1) $(arg frame)" required="true"/>
    <node pkg="rosnao_bridge" type="image_relay" name="image_relay2" output="screen" args="$(arg shm_id2) $(arg res) $(arg topic2) $(arg frame)" required="true"/>
    <!-- Interface with sdk, launches proxy in nao, and put the image in shared memory -->
    <node pkg="rosnao_bridge" type="image_relay.sh" name="image_relay_sdk1" output="screen" args="$(arg nao_ip) $(arg shm_id1) $(arg res) 0" required="true"/>
    <node pkg="rosnao_bridge" type="image_relay.sh" name="image_relay_sdk2" output="screen" args="$(arg nao_ip) $(arg shm_id2) $(arg res) 1" required="true"/>
    
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rosnao_bridge)/rviz/cam_stereo.rviz"/>
</launch>